network.go

```diff
@@ -74,6 +74,7 @@ type svcInfo struct {
 	svcMap     map[string][]net.IP
 	svcIPv6Map map[string][]net.IP
 	ipMap      map[string]string
+	ports      map[string][]servicePorts
```

Comment:
svcInfo.ports will be looked up by the service name and each slice entry is for the individual port names published by the service.